reconcile/utils/vault.py

```diff
@@ -315,7 +315,16 @@ def _write_v1(self, path, data):
 
     def _list(self, path: str) -> dict:
         try:
-            return self._client.list(path)
+            response = self._client.list(path)
+            if response is not None:
+                return response
+            # If vault path is from a v2 engine hvac.list(path) returns None,
+            # to list secrets from a v2 engine we need to use hvac.secrets.kv.v2.list_secrets
```

Comment:
I'm having a hard time finding this feature documented, so I'm not sure we can rely on it.

If you need different methods to list, why don't you first find which is the version of the secret engine and then use the relevant engine methods instead?

Reply:
Documentation about the v2 method is here https://hvac.readthedocs.io/en/latest/usage/secrets_engines/kv_v2.html#list-secrets 

I can try to adapt the code with two different reading methods checking!

Reply:
I meant the main `list` behaviour... I guess that if it tries to use the v1 methods against a v2 secret it returns `None` , but I'd rather try to be more explicit here.